dev/src/dev/db_tracking.clj

```diff
@@ -0,0 +1,124 @@
+(ns dev.db-tracking
+  "A set of utility function to track database changes.
+  Use this when you want to observe changes of database models when doing stuffs on UI.
+
+  How to use this?
+    > (track! models/Dashboard models/Card models/DashboardCard)
+    -- Go on UI and do stuffs like (i.e: update viz-settings of a dashcard).
+
+    > (changes)
+    ;; => {:report_card {:insert ...}}
+
+  You can use [[reset-changes!]] to clear our all the current trackings.
+  And [[untrack-all!]] or [[untrack!]] to stop tracking."
+  (:require
+   [clojure.pprint :as pprint]
+   [metabase.util :as u]
+   [methodical.core :as m]
+   [toucan2.core :as t2]
+   [toucan2.model :as t2.model]
+   [toucan2.tools.before-delete :as t2.before-delete]
+   [toucan2.tools.before-insert :as t2.before-insert]
+   [toucan2.tools.before-update :as t2.before-update]))
+
+(def ^:private tracking (atom {}))
```

Comment:
Suggestion: could this be called `changes`, to make it more consistent with `reset-changes` and `changes`?

Reply:
I would also prefer this to be public, so we can do funky things like add watches to the atom

Reply:
good point, will make it public and change the name